### PR TITLE
CMR-6951 - switched muliti page requests from paging to scroll

### PR DESCRIPTION
**Pull request by jceaser**
A very large change in the search api from page based requests to scrolling. Scrolling is auto created 

Other changes include:
* improved management of dictionary data
* moved a huge chunk of common.search_by_page() to _make_search_request()
* since the reason for moving to scroll was performance, also added experimintal_search_by_page_generator() to utilize python function generators
* fixed logging and made it public to monitor scrolls.

### CODE REVIEW

**Review by mitchstartzel — APPROVED** [review]
Paired for a while with Thomas and reviewed all changed - approved assuming he made all the changes we discussed.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -25,3 +25,9 @@ jobs:
       - name: Run Unit Tests
         run: python3 -m unittest discover
         working-directory: ./CMR/python
+      #- name: Package
```
Comment (jaybarra) [inline .github/workflows/tests.yml:28]: Let's delete commented out sections

File: CMR/python/cmr/search/granule.py [hunk 2]
```diff
@@ -114,9 +135,25 @@ def open_api(section='#granule-search-by-parameters'):
     """
     scom.open_api(section)
 
+def set_logging_to(level):
+    """
+    Set the logging level to the stated value
+    Parameters:
+        level: a value like logging.INFO
```
Comment (jaybarra) [inline CMR/python/cmr/search/granule.py:142]: Let's enumerate valid values here or give a link to where they are defined.

**Review by jaybarra — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (jceaser) [reply .github/workflows/tests.yml]: This file has a merge conflict, I'll do it then (actually I think master version will bring this code in)

File: CMR/python/cmr/search/granule.py (hunk 2 above)
Comment (jceaser) [reply CMR/python/cmr/search/granule.py:144]: can do

Comment (jceaser) [reply CMR/python/cmr/search/granule.py:144]: change made for both granule and collection search code

**Review by jceaser — COMMENTED** [review]
I made the changes requested by Mitch

**Review by jaybarra — APPROVED** [review]

**Pull request merged**